implementations/scripts/Commit.py

```diff
@@ -0,0 +1,90 @@
+import math
+import pandas as pd
+from Metric import Metric
+import utils
+
+
+class Commit(Metric):
+    """
+    Initilizes self.df, the dataframe with one commit per row.
+
+    :param data_list: A list of dictionaries.
+        Each element is a Perceval dictionary, obtained from a JSON
+        file or from Perceval directly.
+
+    :param date_range: A tuple which represents the period of interest
+        It is of the form (since, until), where since and until are strings.
+        Either, or both can be None. If, for example, since is None, that 
+        would mean that all commits from the first commit to the commit who last
+        falls inside the until range will be included.
+
+    :param is_src_code_obj: An object of IsSourceCode.
+        It is used to determine what comprises source code.
+        """
+
+    def __init__(self, data_list, date_range=(None, None), is_src_code_obj=None):
```

Comment:
Do you think you really need to redefine this method for all children? I think the functionality is pretty much the same for all metrics, so I would include most of what you do here in the 'Metric' class. I'm only not sure about `date_range`, but I think all metrics we're defining work only for a date range, so it could go to Metric.

Reply:
Please see my comment above